crates/interpreter/src/valid.rs

```diff
@@ -30,6 +30,78 @@ pub fn validate(binary: &[u8]) -> Result<(), Error> {
 type Parser<'m> = parser::Parser<'m, Check>;
 type CheckResult = MResult<(), Check>;
 
+#[allow(dead_code)] // TODO(dev/fast-interp)
+#[derive(Default, Copy, Clone)]
+pub struct SideTableEntry(u32);
+
+pub struct SideTableEntryView {
+    /// The amount to adjust the instruction pointer by if the branch is taken.
+    pub delta_ip: i32,
+    /// The amount to adjust the side-table pointer by if the branch is taken.
+    pub delta_stp: i32,
+    /// The number of values that will be copied if the branch is taken.
+    pub val_cnt: u32,
+    /// The number of values that will be popped if the branch is taken.
+    pub pop_cnt: u32,
+}
+
+#[allow(dead_code)] // TODO(dev/fast-interp)
+impl SideTableEntry {
+    const DELTA_IP_MASK: u32 = 0x0000ffff;
+    const DELTA_STP_MASK: u32 = 0x003f0000;
+    const VAL_CNT_MASK: u32 = 0x07c00000;
+    const POP_CNT_MASK: u32 = 0xf8000000;
+
+    fn new(view: SideTableEntryView) -> Result<Self, Error> {
+        let mut fields = 0;
+        fields |= bit_field::into_signed_field(Self::DELTA_IP_MASK, view.delta_ip)?;
+        fields |= bit_field::into_signed_field(Self::DELTA_STP_MASK, view.delta_stp)?;
+        fields |= bit_field::into_field(Self::VAL_CNT_MASK, view.val_cnt)?;
+        fields |= bit_field::into_field(Self::POP_CNT_MASK, view.pop_cnt)?;
+
+        Ok(SideTableEntry(fields))
+    }
+
+    fn view(self) -> SideTableEntryView {
+        let delta_ip = bit_field::from_signed_field(Self::DELTA_IP_MASK, self.0);
+        let delta_stp = bit_field::from_signed_field(Self::DELTA_STP_MASK, self.0);
+        let val_cnt = bit_field::from_field(Self::VAL_CNT_MASK, self.0);
+        let pop_cnt = bit_field::from_field(Self::POP_CNT_MASK, self.0);
+
+        SideTableEntryView { delta_ip, delta_stp, val_cnt, pop_cnt }
+    }
+}
+
+mod bit_field {
+    #[allow(unused_imports)]
```

Comment:
The reason of adding it is that there is an unused error for the `use super::Unsupported` below, which is in fact used in `into_field()`. Not sure how to get around it.